rustls/src/client/tls12.rs

```diff
@@ -430,10 +428,7 @@ fn emit_certificate(
 }
 
 fn emit_clientkx(transcript: &mut HandshakeHash, common: &mut CommonState, pub_key: &[u8]) {
-    let mut buf = Vec::new();
-    let ecpoint = PayloadU8::new(Vec::from(pub_key));
-    ecpoint.encode(&mut buf);
-    let pubkey = Payload::new(buf);
+    let pubkey = Payload::new(PayloadU8::new(Vec::from(pub_key)).get_encoding());
```

Comment:
Please avoid making unrelated changes in the same PR or commit.

Reply:
Oh I wasn't to but @ctz suggested it on their review. I'll revert it then.